crates/wdl-engine/src/value.rs

```diff
@@ -2717,37 +2770,172 @@ struct TaskData {
     ext: Object,
 }
 
-/// Represents a value for `task` variables in WDL 1.2.
+/// Represents a `task.previous` value containing requirements from a previous attempt.
+///
+/// The requirements are stored in an `Arc<HashMap>` for cheap cloning.
+#[derive(Debug, Clone)]
+pub struct PreviousRequirementsValue(Arc<HashMap<String, Value>>);
+
+impl PreviousRequirementsValue {
+    /// Creates a new previous requirements value.
+    pub fn new(requirements: HashMap<String, Value>) -> Self {
+        Self(Arc::new(requirements))
+    }
+
+    /// Creates an empty previous requirements value.
+    pub fn empty() -> Self {
```

Comment:
I recognize we could `#[derive(Default)]` here, but I like keeping `empty()` to indicate to the reader in the downstream methods the intention.